src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/ModelMetadata.cs

```diff
@@ -406,7 +406,7 @@ private void InitializeTypeInformation()
                 var enumerableType = ClosedGenericMatcher.ExtractGenericInterface(ModelType, typeof(IEnumerable<>));
                 ElementType = enumerableType?.GenericTypeArguments[0];
 
-                if (ElementType == null && typeof(IEnumerable).IsAssignableFrom(ModelType))
+                if (ElementType == null)
```

Comment:
See [line 393](https://github.com/aspnet/Mvc/pull/4173/files#diff-438d1905c3cf0d136b093a030abe0d41R393). That guarantees the second part of the former condition is `true`.